src/adapter/src/coord/introspection.rs

```diff
@@ -94,9 +94,10 @@ pub fn check_cluster_restrictions(
         return Ok(());
     }
 
-    // Allows explain queries.
-    if let Plan::Explain(_) = plan {
-        return Ok(());
+    // Only continue, and check restrictions, if a Plan would run some computation on the cluster.
+    match plan {
+        Plan::Subscribe(_) | Plan::Peek(_) | Plan::ReadThenWrite(_) => (),
```

Comment:
What about `Plan::CreateMaterializedView`, `Plan::CreateIndex`, `Plan::CreateSource`, `Plan::CreateSink`, and maybe `Plan::CreateClusterReplica`?

Reply:
Those are already restricted because we specify `mz_introspection` as a "read-only" cluster. I added some more tests to `system-cluster.slt` to validate this behavior